src/Blocks/AbstractBlocks.php

```diff
@@ -486,6 +542,95 @@ protected function prepareComponentAttributes(array $blockDetails): array
 		return $output;
 	}
 
+	/**
+	 * Build components dependency tree for blocks.
+	 *
+	 * @param array $blocks List of all blocks.
+	 * @param array $components List of all components.
+	 *
+	 * @return array
+	 */
+	private function buildDependencyBlocksTree(array $blocks, array $components): array
+	{
+		$output = [];
+
+		foreach ($blocks as $item) {
+			$blockName = $item['blockName'] ?? '';
+			$componentsItem = $item['components'] ?? [];
+
+			$output[$blockName] = array_unique($this->buildDependencyComponentsInnerTree($componentsItem, $components));
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Build components dependency tree for blocks.
+	 *
+	 * @param array $components List of all blocks.
+	 *
+	 * @return array
+	 */
+	private function buildDependencyComponentsTree(array $components): array
+	{
+		$output = [];
+
+		foreach ($components as $item) {
+			$componentName = $item['componentName'] ?? '';
+			$componentsItem = $item['components'] ?? [];
+
+			if (! $componentName) {
+				continue;
+			}
+
+			$output[$componentName] = array_unique($this->buildDependencyComponentsInnerTree($componentsItem, $components));
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Build inner recursive dependency tree for components.
+	 *
+	 * @param array $componentsList List of components to check.
+	 * @param array $components List of all components.
+	 *
+	 * @return array
+	 */
+	private function buildDependencyComponentsInnerTree(array $componentsList, array $components): array
+	{
+		$output = [];
+
+		if (! $componentsList) {
+			return $output;
+		}
+
+		foreach ($componentsList as $key => $value) {
+			$items = array_values(
+				array_filter(
+					$components,
+					function ($item) use ($value) {
+						$componentName = $item['componentName'] ?? '';
+
+						if ($componentName === $value) {
+							return $item;
+						}
```

Comment:
This is not the correct usage of `array_filter`. The `array_filter` will go through every item in the array, check the condition, and if the condition is `true` it will keep the item in the array. If not, it will remove it.

So if your intention is to just remove the items, then the only thing you need here is:

```suggestion
						return $componentName === $value;
```

If you need to return a different thing in the array, then you need `array_map`.

Reply:
Fixed